src/main/java/com/iexec/commons/poco/chain/IexecHubAbstractService.java

```diff
@@ -614,20 +642,28 @@ public Optional<ChainDeal> getChainDeal(String chainDealId) {
 
             final ChainDeal chainDeal = ChainDeal.parts2ChainDeal(chainDealId, deal, app, category, dataset);
 
-            if (chainDeal.getStartTime() == null
-                    || chainDeal.getStartTime().longValue() <= 0) {
-                log.error("Deal start time should be greater than zero (likely a " +
-                                "blockchain issue) [chainDealId:{}, startTime:{}]",
-                        chainDealId, chainDeal.getStartTime());
-                return Optional.empty();
-            }
-            return Optional.of(chainDeal);
+            return validateChainDeal(chainDeal);
         } catch (Exception e) {
             log.error("Failed to get ChainDeal [chainDealId:{}]", chainDealId, e);
         }
         return Optional.empty();
     }
 
+    /**
+     * Checks if deal is valid, i.e. has a positive start time allowing to compute deadlines.
+     *
+     * @param chainDeal The {@code ChainDeal} to check
+     * @return {@code Optional.of(chainDeal)} if valid, {@code Optional.empty()} otherwise
+     */
+    private Optional<ChainDeal> validateChainDeal(final ChainDeal chainDeal) {
+        if (chainDeal.getStartTime() == null || chainDeal.getStartTime().longValue() <= 0) {
+            log.error("Deal start time should be greater than zero (likely a blockchain issue) [chainDealId:{}, startTime:{}]",
+                    chainDeal.getChainDealId(), chainDeal.getStartTime());
+            return Optional.empty();
+        }
+        return Optional.of(chainDeal);
+    }
```

Comment:
With such a name, I would expect a boolean return value. What about the following?
```suggestion
    private boolean validateChainDeal(final ChainDeal chainDeal) {
        if (chainDeal.getStartTime() == null || chainDeal.getStartTime().longValue() <= 0) {
            log.error("Deal start time should be greater than zero (likely a blockchain issue) [chainDealId:{}, startTime:{}]",
                    chainDeal.getChainDealId(), chainDeal.getStartTime());
            return false;
        }
        return true;
    }
```
Which could then be called as follows:
```java
return validateChainDeal(chainDeal)
  ? Optional.of(chainDeal) 
  : Optional.empty();
```

It would make more sense IMHO.

Reply:
It was the first implementation but did not allow to pass the 80% code coverage threshold due to multiple non-tested conditions